lib/marquee.dart

```diff
@@ -514,6 +532,9 @@ class _MarqueeState extends State<Marquee> with SingleTickerProviderStateMixin {
   /// Causes the controller to scroll one round.
   Future<void> _makeRoundTrip() async {
     // Reset the controller, then accelerate, move linearly and decelerate.
+    final canMakeRoundTrip = widget.stopAfterMaxRound == 0 ||
+        _roundCounter < widget.stopAfterMaxRound;
+    if (!canMakeRoundTrip) return;
```

Comment:
I'm not sure this is the right place to put the code – the method is already named `_makeRoundTrip`, so you expect it to actually make one. Instead, what if we put these three lines into the `_scroll` method? The `_scroll` method is used in `initState` and as long as it returns `true`, it's called again. Instead of just returning `_running`, it could return `_running && canMakeRoundTrip`.

Also, we could rename `canMakeRoundTrip` to either `shouldMakeRoundTrip` or, even better, negate it and name it something like `isDone`. Come to think of it, we could define it as a getter field `bool get isDone => ...`, that would make the code inside the methods cleaner.

Reply:
the getter is a better option indeed. I'm applying the changes

Reply:
done